app/main/views/suppliers.py

```diff
@@ -207,13 +238,21 @@ def upload_countersigned_agreement_file(supplier_id, framework_slug):
 @login_required
 @role_required('admin-ccs-sourcing')
 def remove_countersigned_agreement_file(supplier_id, framework_slug):
+    supplier_framework = data_api_client.get_supplier_framework_info(supplier_id, framework_slug)['frameworkInterest']
+    document = supplier_framework.get('countersignedPath')
     agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
-    document = get_agreement_document_path(framework_slug, supplier_id, COUNTERSIGNED_AGREEMENT_FILENAME)
 
     if request.method == 'GET':
         flash('countersigned_agreement', 'remove_countersigned_agreement')
 
     if request.method == 'POST':
+        # Remove path first - as we don't want path to exist in DB with no corresponding file in S3
+        # But an orphaned file in S3 wouldn't be so bad
+        data_api_client.update_framework_agreement(
+            supplier_framework['agreementId'],
+            {"countersignedAgreementPath": None},
+            current_user.email_address
+        )
```

Comment:
If this all sounds good @pcraig3  and you are happy with the rest of your comments being resolved then can we have a thumbs up pretty please...